agent-control/src/bin/main_onhost.rs

```diff
@@ -16,8 +15,38 @@ use std::error::Error;
 use std::process::ExitCode;
 use tracing::{error, info, trace};
 
+#[cfg(target_os = "windows")]
+use newrelic_agent_control::command::windows::{WINDOWS_SERVICE_NAME, setup_windows_service};
+
+#[cfg(target_os = "windows")]
+windows_service::define_windows_service!(ffi_service_main, service_main);
+
 fn main() -> ExitCode {
-    Command::run(AGENT_CONTROL_MODE_ON_HOST, _main)
+    #[cfg(not(target_os = "windows"))]
+    {
+        Command::run(AGENT_CONTROL_MODE_ON_HOST, _main)
+    }
```

Comment:
Total nit but I'd prefer to have anything not Windows nor Unix just fail to compile, so I'd use `#[cfg(target_family = "unix")]` here instead of not-windows. I think we have it like this in the process management.